File: ga/gpt3_5_result_2024_6/result/1968B_gpt/26_1968B.c
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut input = stdin.lock();
    
    let mut input_line = String::new();
    input.read_line(&mut input_line).unwrap();
    let t: i32 = input_line.trim().parse().unwrap();
    
    for _ in 0..t {
        input_line.clear();
        input.read_line(&mut input_line).unwrap();
        let parts: Vec<i32> = input_line.split_whitespace()
                                    .map(|x| x.parse().unwrap())
                                    .collect();
        let n = parts[0] as usize;
        let m = parts[1] as usize;
        
        input_line.clear();
        input.read_line(&mut input_line).unwrap();
        let sn = input_line.trim();
        
        input_line.clear();
        input.read_line(&mut input_line).unwrap();
        let sm = input_line.trim();
        
        let mut i = 0;
        let mut j = 0;
        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }
        
        println!("{}", i);
    }
}

File: ga/glm4_result_2024_6/result/1968B_cpp_glm/26_1968B.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: usize = input.trim().parse().expect("Invalid input");

    for _ in 0..t {
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read line");
        let mut values = input.trim().split_whitespace();
        let x: i32 = values.next().unwrap().parse().expect("Invalid input");
        let k: i32 = values.next().unwrap().parse().expect("Invalid input");
        let a: String = values.next().unwrap().to_string();
        let b: String = values.next().unwrap().to_string();

        let mut ans = 0;
        for (i, &ch) in b.chars().enumerate() {
            if ans < a.len() && a.chars().nth(i).unwrap() == ch {
                ans += 1;
            }
        }

        println!("{}", ans);
    }
}